joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/testfixtures/JavaSrcCodeToCpgFixture.scala

```diff
@@ -23,31 +22,25 @@ trait JavaSrcFrontend extends LanguageFrontend {
   }
 }
 
-class JavaSrcTestCpg(enableTypeRecovery: Boolean = false) extends TestCpg with JavaSrcFrontend {
-  private var _withOssDataflow = false
-
-  def withOssDataflow(value: Boolean = true): this.type = {
-    _withOssDataflow = value
-    this
-  }
+class JavaSrcTestCpg(enableTypeRecovery: Boolean = false) extends TestCpg with JavaSrcFrontend with SemanticTestCpg {
 
   override def applyPasses(): Unit = {
-    X2Cpg.applyDefaultOverlays(this)
+    super.applyPasses()
     if (enableTypeRecovery) JavaSrc2Cpg.typeRecoveryPasses(this).foreach(_.createAndApply())
-    if (_withOssDataflow) {
-      val context = new LayerCreatorContext(this)
-      val options = new OssDataFlowOptions()
-      new OssDataFlow(options).run(context)
-    }
   }
 
 }
 
-class JavaSrcCode2CpgFixture(withOssDataflow: Boolean = false, enableTypeRecovery: Boolean = false)
-    extends Code2CpgFixture(() => new JavaSrcTestCpg(enableTypeRecovery).withOssDataflow(withOssDataflow)) {
+class JavaSrcCode2CpgFixture(
+  withOssDataflow: Boolean = false,
+  extraFlows: List[FlowSemantic] = List.empty,
+  enableTypeRecovery: Boolean = false
+) extends Code2CpgFixture(() =>
+      new JavaSrcTestCpg(enableTypeRecovery).withOssDataflow(withOssDataflow).withExtraFlows(extraFlows)
```

Comment:
interesting formatting... did scalafmt come up with that? in that case we need to update our config i guess... :)